sympy/functions/special/gamma_functions.py

```diff
@@ -799,11 +799,26 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
 
     def fdiff(self, argindex=2):
         if argindex == 2:
-            n, z = self.args[:2]
+            n, z = self.args
             return polygamma(n + 1, z)
         else:
             raise ArgumentIndexError(self, argindex)
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        s, z = self.args
+        d = z.subs(x, 0) # expansion point
+        e = (z-d).as_leading_term(x).as_coeff_exponent(x)[1]
+        if s.is_integer and s.is_nonnegative and \
+                d.is_integer and d.is_nonpositive and e.is_positive:
+            terms = [(-1)**(s+1) * factorial(s) / (z-d)**(s+1)]
+            for k in range(ceiling(n/e)):
```

Comment:
Is it possible that `n/e` here could be symbolic?

Reply:
It is certainly possible in the series methods for Bessel functions, even after the extensive changes of #23844:

```python
from sympy import *
x = symbols("x")
a = symbols("a", positive=True)
besseli(0, x**a).series(x, 0, 12)
# newn, the number of terms to use, is calculated here as the symbolic ceiling(12/a)
# ... NotImplementedError: Not sure of sign of 12 - a
```

I used the new Bessel series code as a model and as such see no need to ensure that `n/e` is numeric.